services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -67,8 +67,15 @@ def determine_feedback_from_history(feedback_history)
     end
 
     def regex_is_passing?(entry)
-      regex_rules.none? do |regex_rule|
-        regex_rule.entry_failing?(entry)
+      return true if regex_rules.empty?
+      if regex_rules.first.incorrect_sequence?
+        regex_rules.none? do |regex_rule|
```

Comment:
`any?` and `none?` relate inversely, so you only need to iterate over `regex_rules` once, then memoize the result and evaluate it in both branches of the conditional.

Reply:
Pushed an update based on our conversation! (Instead of memoizing the result of one loop, kept both loops but defined them under more clear and reasonable names).